src/modules/configurator/hooks/useSavedConfiguration.tsx

```diff
@@ -0,0 +1,86 @@
+import { useState, useEffect } from "react";
+import { useRecoilValue } from "recoil";
+import { userConfigurationsState } from "../state";
+import { Timestamp } from "firebase/firestore";
+import { monthNames } from "../components/const";
+import { CarImageFetchProps, CarModel, CarPosition } from "../types";
+import { fetchCarImageByColorAndVariant } from "../services";
+
+interface Props {
+  name: CarModel;
+  colorLabel: string;
+  wheelsLabel: number;
+  creationDate: Timestamp;
+}
+
+export function useSavedConfiguration({
+  name,
+  colorLabel,
+  wheelsLabel,
+  creationDate,
+}: Props) {
+  const [imgUrl, setImgUrl] = useState("");
+  const [isOptionsDropdownOpen, setIsOptionsDropdownOpen] = useState(false);
+  const userConfigurations = useRecoilValue(userConfigurationsState);
+
+  const nthFormat = (day: number) => {
+    if (day > 3 && day < 21) return "th";
+    switch (day % 10) {
+      case 1:
+        return "st";
+      case 2:
+        return "nd";
+      case 3:
+        return "rd";
+      default:
+        return "th";
+    }
+  };
```

Comment:
For date formating, we usually use `date-fns` package, you can check it out later